### PR TITLE
add support config TLS connection

### PR DESCRIPTION
**Pull request by sebasrock**
Hi guys 

I added a client option called  `WithTLS` where it's possible to add configuration in order to get a `TLS` connection with the MQTT broker, also I changed the method `WithTCPAddress` to `WithAddress` such the communication now can be TLS too

### CODE REVIEW

**Comment by ajatprabha** [comment]
The change is backwards incompatible as you're renaming the Option. I'd explore if this can be done in a non-breaking way. Do you explicitly need to rename the Option because it can also work without that?

**Comment by sebasrock** [comment]
Hi @ajatprabha, how you say I can work without change the Options, I am going to update the PR 
🙇‍♂️

File: client.go [hunk 1]
```diff
@@ -132,9 +132,16 @@ func toClientOptions(c *Client, o *clientOptions) *mqtt.ClientOptions {
 		opts.SetClientID(o.clientID)
 	}
 
+	if o.tlsConfig != nil {
+		o.brokerAddress = fmt.Sprintf("tls://%s", o.brokerAddress)
+	} else {
```
Comment (ajatprabha) [inline client.go:137]: `else` pattern can be avoided, let's move this to its own function to format the address.

File: client_options.go [hunk 2]
```diff
@@ -94,7 +102,7 @@ func WithOnReconnect(handler OnReconnectHandler) ClientOption {
 // Default values for hostname is "127.0.0.1" and for port is 1883.
 func WithTCPAddress(host string, port uint16) ClientOption {
```
Comment (ajatprabha) [inline client_options.go:103]: We can add another `Option` named `WithAddress` and mark this one as `DEPRECATED` and add a comment to use the new `Option`.

**Review by ajatprabha — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (sebasrock) [reply client.go]: Fixed 🙇

File: client_options.go (hunk 2 above)
Comment (sebasrock) [reply client_options.go:106]: Fixed 🙇

File: client_options.go [hunk 3]
```diff
@@ -92,9 +100,22 @@ func WithOnReconnect(handler OnReconnectHandler) ClientOption {
 
 // WithTCPAddress sets the broker address to be used.
 // Default values for hostname is "127.0.0.1" and for port is 1883.
+//
+// Deprecated: This function used to work just with connections TCP,
```
Comment (ajatprabha) [inline client_options.go:104]: nit: 
```go
// Deprecated: This Option used to work with plain TCP connections,
// it's now possible to use TLS with WithAddress and WithTLS combination.
```

File: client.go [hunk 4]
```diff
@@ -148,6 +149,15 @@ func toClientOptions(c *Client, o *clientOptions) *mqtt.ClientOptions {
 	return opts
 }
 
+func formatAddressWithProtocol(opts *clientOptions) string {
+	switch {
```
Comment (ajatprabha) [inline client.go:153]: switch statement can be avoided
```go
func formatAddressWithProtocol(opts *clientOptions) string {
	if opts.tlsConfig != nil {
		return fmt.Sprintf("tls://%s", opts.brokerAddress)
	}

	return fmt.Sprintf("tcp://%s", opts.brokerAddress)
}
```

File: client_options.go [hunk 5]
```diff
@@ -92,9 +100,22 @@ func WithOnReconnect(handler OnReconnectHandler) ClientOption {
...
+// now it's possible use TLS too. Use courier/WithAddress instead.
 func WithTCPAddress(host string, port uint16) ClientOption {
 	return func(o *clientOptions) {
-		o.brokerAddress = fmt.Sprintf("tcp://%s:%d", host, port)
+		o.brokerAddress = fmt.Sprintf("%s:%d", host, port)
+	}
+}
+
+// WithAddress sets the broker address to be used.
```
Comment (ajatprabha) [inline client_options.go:112]: nit:
```go
// WithAddress sets the broker address to be used.
// To establish a TLS connection, use WithTLS Option along with this.
// Default values for hostname is "127.0.0.1" and for port is 1883.
```

**Review by ajatprabha — CHANGES_REQUESTED** [review]

File: client_options.go (hunk 3 above)
Comment (sebasrock) [reply client_options.go]: Fixed 🙇

File: client.go (hunk 4 above)
Comment (sebasrock) [reply client.go]: Fixed 🙇

File: client_options.go (hunk 5 above)
Comment (sebasrock) [reply client_options.go:112]: Fixed 🙇

**Review by ajatprabha — APPROVED** [review]

**Comment by ajatprabha** [comment]
@sebasrock You need to run `make docs` to fix the failing test stage.

**Comment by sebasrock** [comment]
thank you @ajatprabha

**Pull request merged**